src/app/routes/api/suggestionHandler.ts

```diff
@@ -49,9 +50,52 @@ export default function(requ : Request, resp : Response) : void {
 	};
 
 	log('Received comment from "%s"', email);
+	return new SuggestionModel(suggest).save();
+}
+
+function getErrorCode (errorCode) {
+	const ERROR_CODES = {
+		'missing-input-secret': 'Unexpected Server Error (1)',
+		'invalid-input-secret': 'Unexpected Server Error (2)',
+		'missing-input-response': 'Missing reCAPTCHA value',
+		'invalid-input-response': 'Invalid reCATPCHA value',
+		'timeout-or-duplicate': 'The validation has timed out and is no longer valid',
+		'bad-request': 'The request is invalid or malformed',
+	};
+	if (Array.isArray(errorCode)) {
+		const errors = errorCode.map(function (code) {
+			return getErrorCode(code);
+		});
+		return errors.join('\r\n');
+	}
+	return ERROR_CODES[errorCode] || (
+		errorCode ? ('Unexpected reCAPTCHA error: ' + errorCode) : 'Unexpected reCAPTCHA error');
+}
 
-	new SuggestionModel(suggest).save().then(() => {
-		okResponse(resp, { sent: true });
-	})
-	.catch(error => errorResponse(resp, error));
+export default function(requ : Request, resp : Response) : void {
+	if (!requ.body.captcha) {
+		const msg = 'Missing captcha parameter';
+		return errorResponse(resp, new Error(msg), msg, {
+			status: 400,
+		});
+	}
+	const secretKey = '6LdTeikUAAAAAIRt_4uVbT3HlZLUrHb3EeeqDtti';
```

Comment:
Now we have a secret key posted to the repository and it will be visible in the history forever. A new key should be generated and kept in our ticket system.